crates/router/src/core/payouts/helpers.rs

```diff
@@ -433,121 +440,246 @@ pub async fn get_or_create_customer_details(
+        key_store,
+        &TransactionData::<()>::Payout(payout_data),
+        routing_data,
+        eligible_connectors,
     )
-    .change_context(errors::ApiErrorResponse::InternalServerError)
-    .attach_printable("Routing algorithm gave invalid connector")?;
-
-    routing_data.routed_through = Some(connector_name);
-
-    Ok(api::PayoutConnectorCallType::Single(connector_data))
+    .await
 }
 
 pub async fn get_default_payout_connector(
     _state: &AppState,
     request_connector: Option<serde_json::Value>,
-) -> CustomResult<api::PayoutConnectorChoice, errors::ApiErrorResponse> {
+) -> CustomResult<api::ConnectorChoice, errors::ApiErrorResponse> {
     Ok(request_connector.map_or(
-        api::PayoutConnectorChoice::Decide,
-        api::PayoutConnectorChoice::StraightThrough,
+        api::ConnectorChoice::Decide,
+        api::ConnectorChoice::StraightThrough,
     ))
 }
 
 pub fn should_call_payout_connector_create_customer<'a>(
     state: &AppState,
-    connector: &api::PayoutConnectorData,
+    connector: &api::ConnectorData,
     customer: &'a Option<domain::Customer>,
     connector_label: &str,
 ) -> (bool, Option<&'a str>) {
     // Check if create customer is required for the connector
-    let connector_needs_customer = state
-        .conf
-        .connector_customer
-        .payout_connector_list
-        .contains(&connector.connector_name);
-
-    if connector_needs_customer {
-        let connector_customer_details = customer.as_ref().and_then(|customer| {
-            get_connector_customer_details_if_present(customer, connector_label)
-        });
-        let should_call_connector = connector_customer_details.is_none();
-        (should_call_connector, connector_customer_details)
-    } else {
-        (false, None)
+    match PayoutConnectors::try_from(connector.connector_name) {
+        Ok(connector) => {
+            let connector_needs_customer = state
+                .conf
+                .connector_customer
+                .payout_connector_list
+                .contains(&connector);
+
+            if connector_needs_customer {
+                let connector_customer_details = customer.as_ref().and_then(|customer| {
+                    get_connector_customer_details_if_present(customer, connector_label)
+                });
+                let should_call_connector = connector_customer_details.is_none();
+                (should_call_connector, connector_customer_details)
+            } else {
+                (false, None)
+            }
+        }
+        _ => (false, None),
     }
 }
 
+pub async fn get_gsm_record(
+    state: &AppState,
+    error_code: Option<String>,
+    error_message: Option<String>,
+    connector_name: Option<String>,
+    flow: String,
+) -> Option<storage::gsm::GatewayStatusMap> {
+    let get_gsm = || async {
+        state.store.find_gsm_rule(
+                connector_name.clone().unwrap_or_default(),
+                flow.clone(),
```

Comment:
Is this flow equivalent to the `Flow` enum from `router_env` crate?

Reply:
If yes, are we not passing the correct flow names to this function?

Reply:
this flow is picked from router_data's flow. For eg,  api::Authorize. 
For payout retries, the flow is hardcoded to "payout_flow" since we use multiple flow within call_connector_payout and can't distinguish them seperately.